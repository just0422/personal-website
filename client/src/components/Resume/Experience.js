import React, {Component} from 'react';
import {Grid, Row, Col} from 'react-bootstrap';
import Moment from 'react-moment';
import axios from 'axios';

import api from 'APIUtils';
import ErrorModal from 'Error';

import 'stylesheets/resume.scss';

export default class Experience extends Component {
  constructor(props) {
    super(props);

    this.state = {
      skills: [],
      comments: [],
      error: null,
    };
  }

  static defaultProps = {
    job: {
      id: -1,
      name: '',
      title: '',
      start: '',
      end: '',
    },
  };

  componentDidMount() {
    let id = this.props.job.id;
    if (id > 0) {
      axios
				.all([
					api.experiences().getSkills(id),
          axios.get('/api/v1/experiences/' + id + '/comments'),
        ])
        .then(
          axios.spread((skillsResponse, commentsResponse) => {
            this.setState({
              skills: skillsResponse.data,
              comments: commentsResponse.data,
            });
          }),
        )
        .catch(err => {
          this.setState({
            error: err,
          });
        });
    }
  }

  render() {
    let job = this.props.job;
    let skills = this.state.skills
      .map(skill => {
        return skill.name;
      })
      .join();
    let comments = this.state.comments.map((comment, i) => {
      return <li key={i}>{comment.content}</li>;
		});

    if (this.state.error) {
		console.log("ErrorModal should render");
      return <ErrorModal component="Experience" error={this.state.error} />;
    } else {
      return (
        <div>
          <Grid>
            <Row>
              <Col xs={6}>
                <div className="section-element-header">
                  <strong>{job.name}</strong>
                </div>
              </Col>
              <Col xs={6} className="section-element-dates">
                <Moment format={'MMM. YYYY'}>{job.start}</Moment> -{' '}
                <Moment format={'MMM. YYYY'}>{job.end}</Moment>
              </Col>
              <Col xs={12} className="section-element-header">
                <em>
                  {job.title} ({skills})
                </em>
              </Col>
              <Col xs={12} className="section-element-comments">
                <ul>{comments}</ul>
              </Col>
            </Row>
          </Grid>
        </div>
      );
    }
  }
}
