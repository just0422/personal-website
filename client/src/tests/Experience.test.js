import React from 'react';
import {shallow} from 'enzyme';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import Experience from '../components/Resume/Experience';

describe('Experience', () => {
  let mock, job, skills, comments;
  beforeEach(() => {
    mock = new MockAdapter(axios);

    job = {
      id: 1,
      name: 'abcd',
      title: 'def',
      start: 'Aug 2017',
      end: 'Sept 2018',
    };

    skills = {
      data: [
        {
          name: 'abc',
          years: '3',
        },
        {
          name: 'def',
          years: '4',
        },
      ],
    };

    comments = {
      data: [
        {
          content: 'abc',
        },
      ],
    };
  });

  it('should render correctly with no props', () => {
    const component = shallow(<Experience />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with job prop', () => {
    mock.onGet('/api/v1/experiences/1/skills').reply(200, {skills: skills});
    mock.onGet('/api/v1/experiences/1/comments').reply(200, {comments: comments});

    const component = shallow(<Experience job={job} />);
    expect(component).toMatchSnapshot();
  });
});
