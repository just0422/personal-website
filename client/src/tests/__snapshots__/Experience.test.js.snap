// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience should render correctly with job prop 1`] = `ShallowWrapper {}`;

exports[`Experience should render correctly with no props 1`] = `ShallowWrapper {}`;
